File: 0137-single-number-ii/0137-single-number-ii.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var singleNumber = function(nums) {
     let n=nums.length;
        let mp ={};
        for(let i=0;i<n;i++){
            mp[nums[i]] = mp[nums[i]]+1 || 1;
        }
     // console.log(mp);
        for(const key in mp){
            if(mp[key] == 1) return Number(key);
        }
    // console.log(mp);
        return -1;
};

File: 0137-single-number-ii/0137-single-number-ii.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        int n=nums.size();
        //Now finding the unique element
        map<int,int> mp;
        for(int i=0;i<n;i++){
            mp[nums[i]]++;
        }
        for(auto x:mp){
            if(x.second == 1) return x.first;
        }
        return -1;
       }
};